search/lib/api/provider.js

```diff
@@ -25,26 +27,33 @@ function convertProvider (event, provider) {
 }
 
 async function getProvider (request, response) {
-  const providerId = request.params.providerId;
-  logger.info(`GET /${providerId}`);
-  const event = request.apiGateway.event;
-  const provider = convertProvider(event, {
-    'provider-id': providerId,
-    'short-name': providerId
-  });
-  await assertValid(schemas.catalog, provider);
-  response.status(200).json(provider);
+  try {
+    const providerId = request.params.providerId;
+    logger.info(`GET /${providerId}`);
+    const event = request.apiGateway.event;
+    const providerList = await cmr.getProviders();
+    const isProvider = providerList.filter(providerObj => providerObj['provider-id'] === providerId);
+    if (isProvider.length === 0) throw new Error(`Provider [${providerId}] not found`);
```

Comment:
- This list is either 98 or 42 depending on if the environment is UAT or Prod, respectively.

- I think since this is the only place we need to validate manually (not as part of the cmr query parameter) we can leave it as part of this function.

Reply:
🤦 I coulda' totally just checked how big the list was myself, sorry! Still getting my bearings. Was wondering what the cost of doing this validation was, but it seems relatively inexpensive.

👍 to leaving it as part of this function.